OracleDatabase/SingleInstance/dockerfiles/18.4.0/runOracle.sh

```diff
@@ -0,0 +1,218 @@
+    echo "Executing user defined scripts"
+  
+    for f in $SCRIPTS_ROOT/*; do
+        case "$f" in
+            *.sh)     echo "$0: running $f"; . "$f" ;;
+            *.sql)    echo "$0: running $f"; echo "exit" | su -p oracle -c "$ORACLE_HOME/bin/sqlplus / as sysdba @$f"; echo ;;
+            *)        echo "$0: ignoring $f" ;;
+        esac
+        echo "";
+    done
+    
+    echo "DONE: Executing user defined scripts"
+    echo "";
+  
+  fi;
+  
+}
+
+########### Move DB files ############
+function moveFiles {
+   if [ ! -d $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID ]; then
+      su -p oracle -c "mkdir -p $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   fi;
+   
+   su -p oracle -c "mv $ORACLE_HOME/dbs/spfile$ORACLE_SID.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   su -p oracle -c "mv $ORACLE_HOME/dbs/orapw$ORACLE_SID $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   su -p oracle -c "mv $ORACLE_HOME/network/admin/listener.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   su -p oracle -c "mv $ORACLE_HOME/network/admin/tnsnames.ora $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/"
+   mv /etc/sysconfig/oracle-xe-18c.conf $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
+
+   cp /etc/oratab $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/
+      
+   symLinkFiles;
+}
+
+########### Symbolic link DB files ############
+function symLinkFiles {
+
+   if [ ! -L $ORACLE_HOME/dbs/spfile$ORACLE_SID.ora ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/spfile$ORACLE_SID.ora $ORACLE_HOME/dbs/spfile$ORACLE_SID.ora
+   fi;
+   
+   if [ ! -L $ORACLE_HOME/dbs/orapw$ORACLE_SID ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/orapw$ORACLE_SID $ORACLE_HOME/dbs/orapw$ORACLE_SID
+   fi;
+   
+   if [ ! -L $ORACLE_HOME/network/admin/listener.ora ]; then
+      ln -sf $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/listener.ora $ORACLE_HOME/network/admin/listener.ora
+   fi;
+   
+   if [ ! -L $ORACLE_HOME/network/admin/tnsnames.ora ]; then
+      ln -sf $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/tnsnames.ora $ORACLE_HOME/network/admin/tnsnames.ora
+   fi;
+   
+   if [ ! -L /etc/sysconfig/oracle-xe-18c.conf ]; then
+      ln -s $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/oracle-xe-18c.conf /etc/sysconfig/oracle-xe-18c.conf
+   fi;
+
+   cp $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/oratab /etc/oratab
+}
+
+########### SIGTERM handler ############
+function _term() {
+   echo "Stopping container."
+   echo "SIGTERM received, shutting down database!"
+  /etc/init.d/oracle-xe-18c stop
+}
+
+########### SIGKILL handler ############
+function _kill() {
+   echo "SIGKILL received, shutting down database!"
+   /etc/init.d/oracle-xe-18c stop
+}
+
+############# Create DB ################
+function createDB {
+   # Auto generate ORACLE PWD if not passed on
+   export ORACLE_PWD=${ORACLE_PWD:-"`openssl rand -hex 8`"}
+   echo "ORACLE PASSWORD FOR SYS AND SYSTEM: $ORACLE_PWD";
+
+   (echo "$ORACLE_PWD"; echo "$ORACLE_PWD";) | /etc/init.d/oracle-xe-18c configure
```

Comment:
I like @whisdom suggestion to print the log output on error:

```suggestion
   (echo "$ORACLE_PWD"; echo "$ORACLE_PWD";) | /etc/init.d/oracle-xe-18c configure > /tmp/XE_DatabaseCreation.log
```